[dev] Some unit tests fail in `ckeditor5-dev-tests` package on Windows

## 📝 Provide detailed reproduction steps (if any)

Run `pnpm test` in `ckeditor5-dev-tests` package on Windows.

### ✔️ Expected result

All tests pass.

### ❌ Actual result

Some tests fail:

```
$ pnpm test

> @ckeditor/ckeditor5-dev-tests@54.3.4 test D:\Projects\ckeditor\ckeditor5-dev\packages\ckeditor5-dev-tests
> vitest run --config vitest.config.js


 RUN  v4.0.18 D:/Projects/ckeditor/ckeditor5-dev/packages/ckeditor5-dev-tests

 ❯ tests/utils/manual-tests/copyassets.js (1 test | 1 failed) 14ms
     × should copy all assets to the build assets directory 13ms
 ✓ tests/utils/getdefinitionsfromfile.js (5 tests) 11ms
 ✓ tests/utils/automated-tests/assertions/equal-markup.js (8 tests) 22ms
 ✓ tests/utils/transformfileoptiontotestglob.js (35 tests) 44ms
 ✓ tests/utils/automated-tests/assertions/attribute.js (6 tests) 10ms
 ✓ tests/utils/automated-tests/treatwarningsaserrorswebpackplugin.js (1 test) 898ms
     ✓ should reassign warnings to errors and not emit the code when errors are present  896ms
 ✓ tests/utils/getrelativefilepath.js (14 tests) 10ms
 ✓ tests/utils/requiredll.js (10 tests) 6ms
 ❯ tests/utils/manual-tests/createserver.js (15 tests | 3 failed) 82ms
     ✓ should start http server 9ms
     × should listen on given port 13ms
     × should listen on 8125 port if no specific port was given 5ms
     × should call the specified callback when the server is running (e.g. to allow running web sockets) 5ms
     ✓ should use "readline" to listen to the SIGINT event on Windows 4ms
       ✓ should handle a request for a favicon (`/favicon.ico`) 5ms
       ✓ should handle a root request (`/`) 17ms
       ✓ should handle a request for a static resource (`*.html`) 3ms
       ✓ should handle a request for a static resource (`*.js`) 3ms
       ✓ should handle a request for a static resource (`*.json`) 3ms
       ✓ should handle a request for a static resource (`*.js.map`) 3ms
       ✓ should handle a request for a static resource (`*.css`) 3ms
       ✓ should handle a request for a static resource (`*.png`) 3ms
       ✓ should handle a request for a static resource (`*.jpg`) 3ms
       ✓ should handle a request for a non-existing resource 3ms
 ✓ tests/utils/manual-tests/compilescripts.js (10 tests) 16ms
 ✓ tests/utils/automated-tests/parsearguments.js (26 tests) 41ms
 ✓ tests/tasks/runautomatedtests.js (10 tests) 2210ms
     ✓ should create an entry file before tests execution  2168ms
 ✓ tests/utils/manual-tests/getwebpackconfig.js (5 tests) 11ms
 ❯ tests/utils/automated-tests/getkarmaconfig.js (11 tests | 1 failed) 60ms
     ✓ should return basic karma config for all tested files 17ms
     × should define proxies to static assets resources 18ms
     ✓ should contain a list of available plugins 3ms
     ✓ should enable webpack watcher when passed the "karmaConfigOverrides" option (execute in Intellij) 3ms
     ✓ should configure coverage reporter 2ms
     ✓ should use `text-summary` reporter for local development 3ms
     ✓ should use `text` reporter on CI 3ms
     ✓ should remove webpack babel-loader if coverage reporter is removed by overrides 3ms
     ✓ should return custom browser launchers with flags (`CHROME_CI`) 2ms
     ✓ should return custom browser launchers with flags (`CHROME_LOCAL`) 2ms
     ✓ should inherit Chrome flags from the headed mode when using the headless version 2ms
 ✓ tests/utils/automated-tests/getwebpackconfig.js (13 tests) 153ms
 ✓ tests/utils/manual-tests/compilehtmlfiles.js (13 tests) 13ms
 ✓ tests/utils/manual-tests/removedir.js (2 tests) 3ms
 ✓ tests/tasks/runmanualtests.js (27 tests) 23ms
 ✓ tests/index.js (3 tests) 4ms

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 5 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  tests/utils/automated-tests/getkarmaconfig.js > getKarmaConfig() > should define proxies to static assets resources
AssertionError: expected [ 'workspace/entry-file.js', { …(5) } ] to deeply equal ArrayContaining{…}

- Expected
+ Received

- ArrayContaining [
+ [
    "workspace/entry-file.js",
-   ObjectContaining {
-     "pattern": StringContaining "ckeditor5-utils/tests/_assets/**/*",
+   {
+     "included": false,
+     "nocache": false,
+     "pattern": "workspace\\workspace\\workspace\\node_modules\\@ckeditor\\ckeditor5-utils\\tests\\_assets\\**\\*",
+     "served": true,
+     "watched": false,
    },
  ]

 ❯ tests/utils/automated-tests/getkarmaconfig.js:121:31
    119|
    120|   expect( karmaConfig.files ).toHaveLength( 2 );
    121|   expect( karmaConfig.files ).toEqual( expect.arrayContaining( [
       |                               ^
    122|    'workspace/entry-file.js',
    123|    expect.objectContaining( {

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/5]⎯

 FAIL  tests/utils/manual-tests/copyassets.js > copyAssets() > should copy all assets to the build assets directory
AssertionError: expected "vi.fn()" to be called with arguments: [ '/build/assets', Any<Object> ]

Received:

  1st vi.fn() call:

  [
-   "/build/assets",
-   Any<Object>,
+   "\\build\\assets",
+   {
+     "recursive": true,
+   },
  ]


Number of calls: 1

 ❯ tests/utils/manual-tests/copyassets.js:34:23
     32|   copyAssets( buildDir );
     33|
     34|   expect( mkdirSync ).toHaveBeenCalledWith( buildDir + '/assets', expect.any( Object ) );
       |                       ^
     35|   expect( copyFileSync ).toHaveBeenCalledTimes( ASSETS.length + 1 );
     36|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/5]⎯

 FAIL  tests/utils/manual-tests/createserver.js > createManualTestServer() > should listen on given port
Error: expect(received).toHaveBeenCalledExactlyOnceWith(expected)

Expected mock function to have been called exactly once, but it was called 0 times
 ❯ tests/utils/manual-tests/createserver.js:60:24
     58|
     59|   expect( server.listen ).toHaveBeenCalledExactlyOnceWith( 8888 );
     60|   expect( loggerStub ).toHaveBeenCalledExactlyOnceWith( '[Server] Server running at http://localhost:8888/' );
       |                        ^
     61|  } );
     62|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/5]⎯

 FAIL  tests/utils/manual-tests/createserver.js > createManualTestServer() > should listen on 8125 port if no specific port was given
Error: expect(received).toHaveBeenCalledExactlyOnceWith(expected)

Expected mock function to have been called exactly once, but it was called 0 times
 ❯ tests/utils/manual-tests/createserver.js:71:24
     69|
     70|   expect( server.listen ).toHaveBeenCalledExactlyOnceWith( 8125 );
     71|   expect( loggerStub ).toHaveBeenCalledExactlyOnceWith( '[Server] Server running at http://localhost:8125/' );
       |                        ^
     72|  } );
     73|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/5]⎯

 FAIL  tests/utils/manual-tests/createserver.js > createManualTestServer() > should call the specified callback when the server is running (e.g. to allow running web sockets)
Error: expect(received).toHaveBeenCalledExactlyOnceWith(expected)

Expected mock function to have been called exactly once, but it was called 0 times
 ❯ tests/utils/manual-tests/createserver.js:79:17
     77|   createManualTestServer( 'workspace/build/.manual-tests', 1234, spy );
     78|
     79|   expect( spy ).toHaveBeenCalledExactlyOnceWith( server );
       |                 ^
     80|  } );
     81|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/5]⎯

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Errors ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

Vitest caught 14 unhandled errors during the test run.
This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:47:25
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:915:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1653:37

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:53:25
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:915:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1653:37

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:64:25
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:915:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1653:37

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:77:25
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:915:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1653:37

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Rejection ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'on')
 ❯ lib/utils/manual-tests/createserver.js:35:22
     33|    } );
     34|
     35|    readlineInterface.on( 'SIGINT', () => {
       |                      ^
     36|     process.emit( 'SIGINT' );
     37|    } );
 ❯ Module.createManualTestServer lib/utils/manual-tests/createserver.js:22:9
 ❯ tests/utils/manual-tests/createserver.js:98:26
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:145:11
 ❯ ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1243:20
 ❯ runWithTimeout ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1209:10
 ❯ runHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1514:51
 ❯ callSuiteHook ../../node_modules/.pnpm/@vitest+runner@4.0.18/node_modules/@vitest/runner/dist/index.js:1520:25

This error originated in "tests/utils/manual-tests/createserver.js" test file. It doesn't mean the error was thrown inside the file itself, but while it was running.
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯


 Test Files  3 failed | 16 passed (19)
      Tests  5 failed | 210 passed (215)
     Errors  14 errors
   Start at  07:49:55
   Duration  4.13s (transform 2.93s, setup 2.47s, import 25.41s, tests 3.63s, environment 3ms)

 ELIFECYCLE  Test failed. See above for more details.
```

### ❓ Possible solution

_If you have ideas, you can list them here. Otherwise, you can delete this section._

## 📃 Other details

* Browser: …
* OS: …
* First affected CKEditor version: …
* Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

--- a/packages/ckeditor5-dev-release-tools/tests/utils/parallelworker.js
+++ b/packages/ckeditor5-dev-release-tools/tests/utils/parallelworker.js
@@ -30,17 +30,16 @@ vi.mock( 'node:worker_threads', () => ( {
 
 describe( 'parallelWorker (worker defined in executeInParallel())', () => {
 	it( 'should execute a module from specified path and pass a package path and task options as arguments', async () => {
+		const postMessage = vi.mocked( parentPort ).postMessage;
+
 		await import( '../../lib/utils/parallelworker.js' );
 
-		// It's needed because `parallelworker` does not export anything. Instead, it processes
-		// an asynchronous loop. We must wait until the current JavaScript loop ends. Adding a new promise at the end
-		// forces it.
-		await new Promise( resolve => {
-			setTimeout( resolve, 100 );
-		} );
+		// Importing the worker starts async work, so wait until it reports completion.
+		await vi.waitFor( () => {
+			expect( postMessage ).toHaveBeenCalledTimes( 2 );
+			expect( postMessage ).toHaveBeenCalledWith( 'done:package' );
+		}, { timeout: 1000 } );
 
-		expect( vi.mocked( parentPort ).postMessage ).toHaveBeenCalledTimes( 2 );
-		expect( vi.mocked( parentPort ).postMessage ).toHaveBeenCalledWith( 'done:package' );
 		expect( vi.mocked( virtual ) ).toHaveBeenCalledTimes( 2 );
 		expect( vi.mocked( virtual ) ).toHaveBeenCalledWith( '/home/ckeditor/packages/a', taskOptions );
 		expect( vi.mocked( virtual ) ).toHaveBeenCalledWith( '/home/ckeditor/packages/b', taskOptions );

--- a/packages/ckeditor5-dev-tests/tests/utils/automated-tests/getkarmaconfig.js
+++ b/packages/ckeditor5-dev-tests/tests/utils/automated-tests/getkarmaconfig.js
@@ -119,7 +119,7 @@ describe( 'getKarmaConfig()', () => {
 		expect( karmaConfig.files ).toEqual( expect.arrayContaining( [
 			'workspace/entry-file.js',
 			expect.objectContaining( {
-				pattern: expect.stringContaining( 'ckeditor5-utils/tests/_assets/**/*' )
+				pattern: expect.stringMatching( /ckeditor5-utils[/\\]tests[/\\]_assets[/\\]\*\*[/\\]\*/ )
 			} )
 		] ) );
 	} );

--- a/packages/ckeditor5-dev-tests/tests/utils/automated-tests/getkarmaconfig.js
+++ b/packages/ckeditor5-dev-tests/tests/utils/automated-tests/getkarmaconfig.js
@@ -8,18 +8,6 @@ import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
 import getWebpackConfigForAutomatedTests from '../../../lib/utils/automated-tests/getwebpackconfig.js';
 import getKarmaConfig from '../../../lib/utils/automated-tests/getkarmaconfig.js';
 
-vi.mock( 'node:path', async () => {
-	const originalModule = await vi.importActual( 'node:path' );
-
-	return {
-		default: {
-			join: vi.fn( ( ...chunks ) => chunks.join( '/' ) ),
-			dirname: vi.fn(),
-			resolve: originalModule.resolve
-		}
-	};
-} );
-
 vi.mock( '../../../lib/utils/automated-tests/getwebpackconfig.js' );
 
 vi.mock( '../../../lib/utils/resolve-path.js', () => ( {
@@ -119,7 +107,7 @@ describe( 'getKarmaConfig()', () => {
 		expect( karmaConfig.files ).toEqual( expect.arrayContaining( [
 			'workspace/entry-file.js',
 			expect.objectContaining( {
-				pattern: expect.stringMatching( /ckeditor5-utils[/\\]tests[/\\]_assets[/\\]\*\*[/\\]\*/ )
+				pattern: expect.stringContaining( path.join( 'ckeditor5-utils', 'tests', '_assets', '**', '*' ) )
 			} )
 		] ) );
 	} );

--- a/packages/ckeditor5-dev-tests/tests/utils/manual-tests/copyassets.js
+++ b/packages/ckeditor5-dev-tests/tests/utils/manual-tests/copyassets.js
@@ -31,7 +31,7 @@ describe( 'copyAssets()', () => {
 		const buildDir = '/build';
 		copyAssets( buildDir );
 
-		expect( mkdirSync ).toHaveBeenCalledWith( buildDir + '/assets', expect.any( Object ) );
+		expect( mkdirSync ).toHaveBeenCalledWith( path.join( buildDir, 'assets' ), expect.any( Object ) );
 		expect( copyFileSync ).toHaveBeenCalledTimes( ASSETS.length + 1 );
 
 		const scriptPath = path.resolve( import.meta.dirname, '..', '..', '..', 'lib', 'utils', 'manual-tests' );
@@ -40,7 +40,7 @@ describe( 'copyAssets()', () => {
 			expect( copyFileSync ).toHaveBeenNthCalledWith(
 				index + 1,
 				path.resolve( scriptPath, assetPath ),
-				path.resolve( buildDir, 'assets', path.basename( assetPath ) )
+				path.join( buildDir, 'assets', path.basename( assetPath ) )
 			);
 		}
 	} );

--- a/packages/ckeditor5-dev-tests/tests/utils/manual-tests/createserver.js
+++ b/packages/ckeditor5-dev-tests/tests/utils/manual-tests/createserver.js
@@ -37,7 +37,11 @@ describe( 'createManualTestServer()', () => {
 			server = createServer( ...theArgs );
 			servers.push( server );
 
-			vi.spyOn( server, 'listen' );
+			vi.spyOn( server, 'listen' ).mockImplementation( ( _port, callback ) => {
+				process.nextTick( callback );
+
+				return server;
+			} );
 
 			return server;
 		} );
@@ -123,11 +127,28 @@ describe( 'createManualTestServer()', () => {
 	} );
 
 	it( 'should try next port when the requested port is in use', async () => {
-		// Occupy the port first.
-		const blockingServer = http.createServer.getMockImplementation()();
+		const originalCreateServer = http.createServer.getMockImplementation();
+		let attempt = 0;
 
-		await new Promise( resolve => {
-			blockingServer.listen( 49555, resolve );
+		vi.mocked( http.createServer ).mockImplementation( ( ...args ) => {
+			const currentServer = originalCreateServer( ...args );
+
+			attempt++;
+
+			if ( attempt === 1 ) {
+				vi.mocked( currentServer.listen ).mockImplementation( () => {
+					process.nextTick( () => {
+						const error = new Error( 'EADDRINUSE: port is already in use' );
+						error.code = 'EADDRINUSE';
+
+						currentServer.emit( 'error', error );
+					} );
+
+					return currentServer;
+				} );
+			}
+
+			return currentServer;
 		} );
 
 		createManualTestServer( 'workspace/build/.manual-tests', 49555 );
@@ -136,9 +157,9 @@ describe( 'createManualTestServer()', () => {
 			expect( loggerStub ).toHaveBeenCalledWith( '[Server] Server running at http://localhost:49556/' );
 		} );
 
+		expect( servers[ 0 ].listen ).toHaveBeenCalledWith( 49555, expect.any( Function ) );
+		expect( servers[ 1 ].listen ).toHaveBeenCalledWith( 49556, expect.any( Function ) );
 		expect( loggerStub ).toHaveBeenCalledWith( '[Server] Port 49555 is in use, trying 49556...' );
-
-		blockingServer.close();
 	} );
 
 	it( 'should reject when a non-EADDRINUSE error occurs', async () => {

--- a/packages/ckeditor5-dev-tests/tests/utils/manual-tests/createserver.js
+++ b/packages/ckeditor5-dev-tests/tests/utils/manual-tests/createserver.js
@@ -31,6 +31,8 @@ describe( 'createManualTestServer()', () => {
 			info: loggerStub
 		} );
 
+		vi.spyOn( process, 'platform', 'get' ).mockReturnValue( 'linux' );
+
 		vi.spyOn( http, 'createServer' ).mockImplementation( ( ...theArgs ) => {
 			server = createServer( ...theArgs );
 			servers.push( server );
@@ -160,7 +162,7 @@ describe( 'createManualTestServer()', () => {
 
 	describe( 'request handler', () => {
 		beforeEach( async () => {
-			createManualTestServer( 'workspace/build/.manual-tests', 49800 );
+			createManualTestServer( 'workspace/build/.manual-tests', 49700 );
 
 			await vi.waitFor( () => {
 				expect( loggerStub ).toHaveBeenCalled();

--- a/packages/ckeditor5-dev-translations/tests/servetranslations.js
+++ b/packages/ckeditor5-dev-translations/tests/servetranslations.js
@@ -30,7 +30,7 @@ describe( 'serveTranslations()', () => {
 	let cwdSpy;
 
 	beforeEach( () => {
-		cwdSpy = vi.spyOn( process, 'cwd' ).mockReturnValue( '/project' );
+		cwdSpy = vi.spyOn( process, 'cwd' ).mockReturnValue( path.join( path.sep, 'project' ) );
 		vi.mocked( fs.existsSync ).mockReturnValue( false );
 		vi.mocked( fs.readFileSync ).mockReset();
 		vi.mocked( rimraf.sync ).mockReset();
@@ -141,7 +141,7 @@ describe( 'serveTranslations()', () => {
 		expect( module.loaders[ 0 ].options.translateSource( 'SOURCE', 'file.ts' ) ).to.equal( 'translated:file.ts' );
 		expect( translationService.translateSource ).toHaveBeenCalledWith( 'SOURCE', 'file.ts' );
 		expect(
-			translationService.loadPackage.mock.calls.some( call => String( call[ 0 ] ).includes( 'packages/ckeditor5-foo/' ) )
+			translationService.loadPackage.mock.calls.some( call => /packages[/\\]ckeditor5-foo[/\\]/.test( String( call[ 0 ] ) ) )
 		).to.equal( true );
 
 		compilation.hooks.optimizeChunkAssets.call( [ { files: [ 'main.js', 'other.js' ] } ] );

